### DISCUSSION TITLE
Menu

### DISCUSSION THREAD

**Discussion by Rmfjd1978** [h2oai/wave#1575] [Q&A]
Is it possible to make a menu like this without using a markup card with html?

![image](https://user-images.githubusercontent.com/98091225/183431784-ee114517-8ff1-46fe-b080-c6f5097858e9.png)

**Comment by mturoci** [comment]
Almost. Everything should be possible with the existing [header card](https://wave.h2o.ai/docs/widgets/content/header#with-links), the only thing missing is a label for [`ui.menu`](https://wave.h2o.ai/docs/widgets/form/menu) (`App's` string in your example) next to the chevron that needs to be added. Feel free to file a feature request.

**Comment by Rmfjd1978** [comment]
With your sugestion how do i make the submenu?
Because i've tried to put a navgroup inside items a it is not possible

**Reply by mturoci** [reply]
`ui.menu` takes `items` which is a list of `ui.command`. Each command has it's own recursive prop called `items` that takes list of `ui.command`s.

Example:

```py
# Form / Menu
# Create a contextual menu component. Useful when you have a lot of links and want to conserve the space.
# #form #menu
# ---
from h2o_wave import main, app, Q, ui

commands = [
    ui.command(name='profile', label='Profile', icon='Contact', items=[
        ui.command(name='preferences', label='Preferences', icon='Settings'),
        ui.command(name='logout', label='Logout', icon='SignOut'),
    ]),
]


@app('/demo')
async def serve(q: Q):
    if not q.client.initialized:
        q.page['example'] = ui.form_card(box='1 1 2 3', items=[ui.menu(items=commands)])
        q.client.initialized = True
    await q.page.save()
```

<img width="365" alt="image" src="https://user-images.githubusercontent.com/64769322/183585231-94edb370-15a0-4f2b-bdfe-747961ebfabc.png">

**Reply by Rmfjd1978** [reply]
yes but i wanted like this:

![image](https://user-images.githubusercontent.com/98091225/183600077-3cab8ad5-856b-4050-b3fc-1b49a3ce665a.png)

```
ui.inlinine([ui.text(content="App's"),
             ui.menu([ui.command(name='#DataKnowledge', label='Dataknowledge'),
                      ui.command(name='#H2o', label='H2O.ai'),
                      ui.command(name='#Anotacao', label='Anotação')])])

```
but it's better to add a parameter to the menu object, just to make the code more simpliest to the users that use it.

**Reply by mturoci** [reply]
Agree, that's what #1577 is for. Using `ui.inline` is the only workaround for now.
